packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -119,10 +120,36 @@ function useInsertionPoint( {
 					meta
 				);
 			} else {
+				const normalizedBlocks = Array.isArray( blocks )
+					? blocks
+					: [ blocks ];
+
+				let _destinationIndex = destinationIndex;
+				let _destinationRootClientId = destinationRootClientId;
+				// Check if it's possible to insert the block in the current location
+				const canInsertBlocks = normalizedBlocks.every( ( block ) =>
+					canInsertBlockType( block.name, _destinationRootClientId )
+				);
+
+				if ( ! canInsertBlocks && normalizedBlocks.length === 1 ) {
+					while (
+						! canInsertBlockType(
+							normalizedBlocks[ 0 ].name,
+							_destinationRootClientId
+						)
+					) {
+						_destinationIndex =
+							getBlockIndex( _destinationRootClientId ) + 1;
+						_destinationRootClientId = getBlockRootClientId(
+							_destinationRootClientId
+						);
+					}
+				}
```

Comment:
This procedure will crash the browser if there really is no place to insert in. This needs to be handled.

Reply:
Should be handled by checking [null on the `_destinationRootClientId`](https://github.com/WordPress/gutenberg/pull/61873/files#diff-16ab90c30e615c2cfba050dda8bfa30642629bb53adcb73195fc5053ad576f30R145-R148)

Reply:
I think this would suffice and treat a fallback. 
```
const _previousDestinationRootClientId = _destinationRootClientId;
... // get index, get new destination
if ( _previousDestinationRootClientId === _destinationRootClientId) {
        break;
}
```

Reply:
The end result should be like:
```
if ( ! canInsertBlocks && normalizedBlocks.length === 1 ) {
                                       // If it's not possible to insert the block in the current 
location, try to find a new location up the tree
                                        while (
                                                ! canInsertBlockType(
                                                        normalizedBlocks[ 0 ].name,
                                                        _destinationRootClientId
                                                )
                                        ) {
                                               // save the index of the previous destination root client ID
                                               const _previousDestinationRootClientId = _destinationRootClientId;
                                                _destinationIndex = getBlockIndex( _previousDestinationRootClientId ) + 1;

                                               // get the parent of the current destination root client ID
                                                _destinationRootClientId = getBlockRootClientId(
                                                        _destinationRootClientId
                                                );

                                               // Break the loop if we searched the whole tree
                                               if (
                                                       _previousDestinationRootClientId ===
                                                       _destinationRootClientId
                                               ) {
                                                       break;
                                               }
                                        }
                                }
```